citar.el

```diff
@@ -1638,30 +1746,29 @@ URL."
   (citar--check-configuration 'citar-library-paths)
   (unless citar-library-paths
     (user-error "Make sure `citar-library-paths' is non-nil"))
-  (let* ((directory (if (cdr citar-library-paths)
-                        (completing-read "Directory: " citar-library-paths)
-                      (car citar-library-paths)))
-         (filepath (expand-file-name citekey directory))
-         (withext (lambda (extension)
-                    (let* ((extension (or extension (read-string "File extension: "))))
-                      (if (string-empty-p extension)
-                          filepath
-                        (concat filepath "." extension))))))
-    (pcase (read-char-choice "Add file from [b]uffer, [f]ile, or [u]rl? " '(?b ?f ?u))
-      (?b
-       (with-current-buffer (read-buffer "Add file buffer: " (current-buffer))
-         (let ((destfile (funcall withext (and buffer-file-name (file-name-extension buffer-file-name)))))
-           (write-file destfile 'confirm))))
-      (?f
-       (let* ((file (read-file-name "Add file: " nil nil t))
-              (destfile (funcall withext (file-name-extension file))))
-         (copy-file file destfile 1)))  ; last arg integer means to confirm before overwriting
-      (?u
-       (let* ((url (read-string "Add file URL: "))
-              ;; TODO: Use Content-Type HTTP response header to guess file extension
-              (ext (url-file-extension url))
-              (destfile (funcall withext (when (> (length ext) 1) (substring ext 1)))))
-         (url-copy-file url destfile 1))))))
+  (unless citar-add-file-sources
+    (user-error "Make sure `citar-add-file-sources' is non-nil"))
```

Comment:
Instead of checking `citar-add-file-sources` here, add a clause to the `citar--check-configuration` function and just call `(citar--check-configuration 'citar-add-file-source)`. We'd like to centralize all such validation in one place.